modules/fscloud/main.tf

```diff
@@ -9,11 +9,16 @@ locals {
       validate_single_site_location = bucket.single_site_location != null ? tobool("KMS encryption can not be added to single site location, therefore it is not supported in fscloud module.") : null,
     }
   ]
+  existing_cos_instance_guid = var.existing_cos_instance_id != null ? element(split(":", var.existing_cos_instance_id), length(split(":", var.existing_cos_instance_id)) - 3) : null
 }
 
-
+data "ibm_resource_instance" "existing_cos_instance_details" {
+  count      = var.existing_cos_instance_id != null ? 1 : 0
+  identifier = local.existing_cos_instance_guid
+}
 
```

Comment:
Are we doing data lookup only for output variables? If yes, we can skip this.

Reply:
@maheshwarishikha thoughtful comment. The root module (line 88) uses four locals to setup outputs. The fscloud submodule would be consistent if it did the same. That logic does not require the data source, so it would not be required here either.

Reply:
@maheshwarishikha @shemau The root module takes care of `id` and `guid` perfectly using the local but in case of `name` if it is not creating a cos instance then it is outputting `null`. In FSCloud module if we are passing `existing_cos_instance_id` then how to fetch the `name` of that `cos_instance` without using data lookup?

Reply:
If user is providing `existing_cos_instance_id`, then no need to show related details. We can output as null.

Reply:
I have resolved all the comments based on the input but I do think that outputting `null` will create unnecessary confusion for modules which will be consuming FSCloud COS output in future.